docs/ga-roadmap.md

```diff
@@ -0,0 +1,114 @@
+* `config`
+  * Core `config` including service definition, internal components definition will be stabilized.
+  * To be determined which config helpers will be marked as stable (e.g. configgrpc, etc.).
+* `component`
+  * Interfaces and utils to build a Collector internal component (receiver, processor, exporter, extension).
+* `obsreport`
+  * Focus on the public API of this package. It is out of scope to ensure stability for the metrics emitted (focus in phase 2).
+* `service`
+  * Public API to construct a OpenTelemetry Collector Service.
+
+**Action Items:**
+
+* Create a new milestone for this phase, create issues for all the other action items and add them to the milestone.
+* Agreement on all packages/modules that will be marked as stable during this phase.
+* Write a version doc as per [version and stability document](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md).
+  * Previously it was discussed that for the Collector it is fine to release stable golang modules that contain APIs marked as experimental.
+  * Define status schema (experimental/stable), what are they applicable to every module.
+* Investigate if splitting into smaller, more granular, modules is possible.
+  * Define the modules schema, try to not break everyone. See [here](https://github.com/golang/go/wiki/Modules#is-it-possible-to-add-a-module-to-a-multi-module-repository).
+  * Investigate how can we release multiple golang modules from the same repo, without asking people that consume them to use a replace statement. See problem in contrib.
+  * Investigate how to release testonly utils.
+* Review all public APIs and godocs for modules that we want to release in this phase.
+  * Fix all critical issues, and remove unnecessary (“When in doubt leave it out”) public APIs.
+  * Remove all already deprecated code from the stable modules.
+* Transition to opentelemetry-go trace library from opencensus?
+* Investigate if any config helper needs to be released as stable, if any do the review of the public API, godoc and configuration.
+* Investigate tools that check for API compatibility for go modules, enable them for modules that we mark as stable.
+
+### Phase 2
+
+**Tentative Date:** 2021-04-31
+
+**KR:** At the end of this phase the Collector’s end-to-end support for OTLP traces only be marked as GA.
+
+At the end of this phase we want to ensure that the Collector can be run in production, it can receive OTLP trace traffic and emit OTLP trace traffic. The complete list of the packages/modules will be finalized during the first part of this phase, but the tentative list is:
+
+* `receiver`
+  * `receiverhelper` - without scraper utils in this phase.
+  * `otlp`
+* `processor`
+  * `processorhelper`
+  * `batch`
+  * `memory_limiter`
+* `exporter`
+  * `exporterhelper`
+  * `otlp`
+  * `otlphttp`
+* `extension`
+  * `extensionhelper`
+  * `healthcheck`
```

Comment:
pprof? zpages?

Reply:
What would the driver for not including pprof or zpages? Do we need to improve testing?

Reply:
I think those would be realistically in phase 3. Don't want to commit too much, already a lot of work to do.

Reply:
Fair point.